### PR TITLE
Add config.ui.isDisabled option

### PR DESCRIPTION
**Pull request by timleslie**
In particular, this is needed for our testing framework, where we want to spin up an express server for just the graphQL API.

### CODE REVIEW

**Review by mitchellhamilton — COMMENTED** [review]
It should be `isDisabled` as per #461, no?

**Comment by timleslie** [comment]
> It should be `isDisabled` as per #461, no?

Good point, it 💯 should. Will change 👍

File: packages-next/keystone/src/scripts/build/build.ts [hunk 1]
```diff
@@ -16,11 +16,15 @@ export async function build({ dotKeystonePath, projectAdminPath }: StaticPaths)
   console.log('✨ Generating graphQL schema');
   await saveSchemaAndTypes(graphQLSchema, keystone, dotKeystonePath);
 
-  console.log('✨ Generating Admin UI code');
-  await generateAdminUI(config, graphQLSchema, keystone, projectAdminPath);
+  if (config.ui?.isDisabled) {
+    console.log('✨ Skipping Admin UI build');
```
Comment (mitchellhamilton) [inline packages-next/keystone/src/scripts/build/build.ts:20]: This is problematic now that I think about it. The build also builds the API so we can't just skip it like this.

**Comment by timleslie** [comment]
Currently blocked on #4867

File: packages-next/keystone/src/scripts/build/build.ts [hunk 2]
```diff
@@ -23,21 +42,52 @@ export function serializePathForImport(path: string) {
...
+        {
+          mode: 'copy' as const,
+          inputPath: Path.join(pkgDir, 'static', 'tsconfig.json'),
+          outputPath: 'tsconfig.json',
+        },
+        {
+          mode: 'write' as const,
+          src: `import React from 'react'; export default () => <></>`,
+          outputPath: Path.join('pages', 'index.js'),
+        },
```
Comment (mitchellhamilton) [inline packages-next/keystone/src/scripts/build/build.ts:86]:
```suggestion
```

I just tested this locally without the generation of this file and it works so I don't think it's necessary.

Also a nit: `files.push(...[stuff,stuff])` -> `files.push(stuff,stuff)`

File: packages-next/keystone/src/scripts/run/start.ts [hunk 3]
```diff
@@ -28,7 +28,11 @@ export const start = async ({ dotKeystonePath, projectAdminPath }: StaticPaths)
     false,
     projectAdminPath
   );
-  console.log(`👋 Admin UI and graphQL API ready`);
+  if (config.ui?.isDisabled) {
+    console.log(`👋 GraphQL API ready`);
+  } else {
+    console.log(`👋 Admin UI and graphQL API ready`);
```
Comment (mitchellhamilton) [inline packages-next/keystone/src/scripts/run/start.ts:34]:
```suggestion
    console.log(`👋 Admin UI and GraphQL API ready`);
```

File: packages-next/keystone/src/scripts/build/build.ts (hunk 2 above)
Comment (timleslie) [reply packages-next/keystone/src/scripts/build/build.ts:86]: Hmm, I had issues without this file. Let me double check 🤔 

nit: 👍

Comment (timleslie) [reply packages-next/keystone/src/scripts/build/build.ts:86]: OK, have confirmed that `index.js` isn't needed 👍

**Review by mitchellhamilton — APPROVED** [review]

**Pull request merged**